From 0d2dc8d3f977d0dec4272f1867613cc79d2aa7df Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:04:09 -0400
Subject: [usb] xhci: replace USB_MAXINTERFACES with config->desc.bNumInterface

Message-id: <1410887463-20674-178-git-send-email-dzickus@redhat.com>
Patchwork-id: 95492
O-Subject: [RHEL7 PATCH 177/591] xhci: replace USB_MAXINTERFACES with config->desc.bNumInterface
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 64ba419b7a5acf5fc934bfad9cc46c7f17ab8026
Author: Xenia Ragiadakou <burzalodowa@gmail.com>
Date:   Mon Aug 26 23:29:46 2013 +0300

    xhci: replace USB_MAXINTERFACES with config->desc.bNumInterface

    This patch replaces USB_MAXINTERFACES with config->desc.bNumInterface in
    the termination condition for the loop that updates the LPM timeout of the
    endpoints on the cofiguration's interfaces, in xhci_calculate_lpm_timeout(),
    to avoid unnecessary loop cycles since most configurations come with 1-2
    interfaces while USB_MAXINTERFACES is 32.

    Signed-off-by: Xenia Ragiadakou <burzalodowa@gmail.com>
    Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/xhci.c b/drivers/usb/host/xhci.c
index a96b35c..1798c53 100644
--- a/drivers/usb/host/xhci.c
+++ b/drivers/usb/host/xhci.c
@@ -4455,7 +4455,7 @@ static u16 xhci_calculate_lpm_timeout(struct usb_hcd *hcd,
  if (!config)
   return timeout;
 
- for (i = 0; i < USB_MAXINTERFACES; i++) {
+ for (i = 0; i < config->desc.bNumInterfaces; i++) {
   struct usb_driver *driver;
   struct usb_interface *intf = config->interface[i];
 
-- 
1.7.1